packages/flutter_tools/test/general.shard/build_system/targets/localizations_test.dart

```diff
@@ -0,0 +1,118 @@
+      const FakeCommand(
+        command: <String>[
+          'dart',
+          'dev/tools/localization/bin/gen_l10n.dart',
+          '--arb-dir=arb',
+          '--template-arb-file=example.arb',
+          '--output-localization-file=bar',
+          '--untranslated-messages-file=untranslated',
+          '--output-class=Foo',
+          '--header-file=header',
+          '--header=HEADER',
+          '--use-deferred-loading',
+          '--preferred-supported-locales=en_US'
+        ],
+      ),
+    ]);
+    final Directory arbDirectory = fileSystem.directory('arb')
+      ..createSync();
+    arbDirectory.childFile('foo.arb').createSync();
+    arbDirectory.childFile('bar.arb').createSync();
+
+    final LocalizationOptions options = LocalizationOptions(
+      header: 'HEADER',
+      headerFile: Uri.file('header'),
+      arbDirectory: Uri.file('arb'),
+      deferredLoading: true,
+      outputClass: 'Foo',
+      outputLocalizationsFile: Uri.file('bar'),
+      preferredSupportedLocales: 'en_US',
+      templateArbFile: Uri.file('example.arb'),
+      untranslatedMessagesFile: Uri.file('untranslated'),
+    );
+    final Depfile depfile = await generateLocalizations(
+      options: options,
+      logger: logger,
+      fileSystem: fileSystem,
+      processManager: processManager,
+      projectDir: fileSystem.currentDirectory,
+      dartBinaryPath: 'dart',
+      flutterRoot: '',
+    );
+
+    expect(depfile.inputs.map((File file) => file.uri), unorderedEquals(<Uri>[
+      Uri.parse('arb/foo.arb'),
+      Uri.parse('arb/bar.arb'),
+      Uri.parse('file:///header'),
+    ]));
+    expect(depfile.outputs.map((File file) => file.uri), unorderedEquals(<Uri>[
+      Uri.parse('file:///untranslated'),
+      Uri.parse('bar'),
+    ]));
+  });
+
+  testWithoutContext('parseLocalizationsOptions handles valid yaml', () async {
+    final FileSystem fileSystem = MemoryFileSystem.test();
+    final File configFile = fileSystem.file('l10n.yaml')
+      ..writeAsStringSync('''
+arb-dir: arb
+template-arb-file: example.arb
+output-localization-file: bar
+untranslated-messages-file: untranslated
+output-class: Foo
+header-file: header
+header: HEADER
+use-deferred-loading: true
+preferred-supported-locales: en_US
+''');
+
+    final LocalizationOptions options = parseLocalizationsOptions(
+      file: configFile,
+      logger: BufferLogger.test(),
+    );
+
+    expect(options.arbDirectory, Uri.parse('arb'));
+    expect(options.templateArbFile, Uri.parse('example.arb'));
+    expect(options.outputLocalizationsFile, Uri.parse('bar'));
+    expect(options.untranslatedMessagesFile, Uri.parse('untranslated'));
+    expect(options.outputClass, 'Foo');
+    expect(options.headerFile, Uri.parse('header'));
+    expect(options.header, 'HEADER');
+    expect(options.deferredLoading, true);
+    expect(options.preferredSupportedLocales, 'en_US');
+  });
+
+  testWithoutContext('parseLocalizationsOptions throws exception on invalid yaml', () async {
```

Comment:
Maybe clarify "invalid yaml" here. My first thought was it was testing for invalid/valid yaml syntax, but realized later on that it was testing for the validity of the values passed into the yaml files. So, something like: 
```suggestion
  testWithoutContext('parseLocalizationsOptions throws exception on invalid yaml configuration', () async {
```

Reply:
Done